Equality defined on multi-value maps and associated tests

ColmBhandal opened the pull request:
## Description

Defined equality on multi-value maps and associated tests. Also refactored type parameters to be more descriptive.

ColmBhandal committed:
Equality defined on multi-value maps and associated tests

Also refactored type parameters to be more descriptive

ColmBhandal committed:
Updated change log

hctrdev commented on CsharpExtras/Dictionary/MultiValueMapImpl.cs:
```diff
@@ -126,5 +126,43 @@ IEnumerator IEnumerable.GetEnumerator()
             return _setValuedMap.GetEnumerator();
         }
+
+        public override bool Equals(object obj)
```
Check that the maps are the same size first

hctrdev commented on CsharpExtras/Dictionary/MultiValueMapImpl.cs:
```diff
@@ -126,5 +126,43 @@ IEnumerator IEnumerable.GetEnumerator()
+                    }
+                    ISet<TVal>? otherSet = otherMap[key];
+                    ISet<TVal>? thisSet = pair.Value;
+                    bool otherNull = otherSet == null;
```
Rename to `isOtherNull`

hctrdev commented on CsharpExtrasTest/Dictionary/MultiValueMapTest.cs:
```diff
@@ -10,6 +10,52 @@ namespace CsharpExtrasTest.Dictionary
     [TestFixture]
     class MultiValueMapTest
```
Add new test to ensure sets of different sizes return false.

ColmBhandal committed:
Added check for map size inequality as a cheap early break condition

Added a test for maps of different lengths, confirming they're not equal
Renamed a boolean variable using the is prefix convention
